addEventResult/index.ts

```diff
@@ -0,0 +1,65 @@
+import { AzureFunction, Context, HttpRequest } from '@azure/functions';
+import getConnection from '../shared/services/db';
+import { FunctionError, httpErrorHandler, ObjectNotFoundFunctionError, ValidationFunctionError } from '../shared/error/error';
+import EventModel from '../shared/event/event.model';
+import addEventInterviewerSchema from './joi';
+import HttpTriggerResult from '../shared/httpTriggerResult';
+import { Role, User } from '../shared/user/user.interface';
+import { Event, EventResult } from '../shared/event/event.interface';
+import authWrapper from '../shared/authWrapper';
+
+const functionName = 'addEventResult';
+
+const updateEventResults = (event: Event, eventResult: EventResult): EventResult => {
+    let { results } = event;
+    if (!results) {
+        results = {
+            notes: [],
+            questionnaire: [],
+            videoUrl: '',
+        };
+    }
+    if (eventResult.notes && eventResult.notes.length > 0) {
+        results.notes = results.notes.concat(eventResult.notes);
+    }
+    if (eventResult.questionnaire && eventResult.questionnaire.length > 0) {
+        results.questionnaire = results.questionnaire.concat(eventResult.questionnaire);
+    }
+    if (eventResult.videoUrl && eventResult.videoUrl !== '') {
+        results.videoUrl = eventResult.videoUrl;
+    }
+
+    return results;
+};
+
+const addEventResult: AzureFunction = async (context: Context, req: HttpRequest, user: User): Promise<HttpTriggerResult> => {
+    try {
+        const { error } = addEventInterviewerSchema.validate(req);
+        if (error) {
+            throw new ValidationFunctionError(functionName, error.message);
+        }
+
+        if (user.role !== Role.Interviewer) {
+            throw new FunctionError(functionName, 'Cannot add event result for now');
+        }
+
+        await getConnection();
+
+        const eventId = context.bindingData.id;
+        const event = await EventModel.findOne({ _id: eventId, interviewersIds: user.id }).exec();
+        if (!event) {
+            throw new ObjectNotFoundFunctionError(functionName, 'Event', eventId);
+        }
+
+        const { eventResult } = req.body;
```

Comment:
```suggestion
        const { result } = req.body;
```
I think it's more clear